ddtrace/tracer/stats.go

```diff
@@ -0,0 +1,332 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package tracer
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"google.golang.org/protobuf/proto"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+
+	"github.com/DataDog/sketches-go/ddsketch"
+)
+
+type statsPayload struct {
+	Hostname string
+	Env      string
+	Version  string
+	Stats    []statsBucket
+}
+
+type statsBucket struct {
+	Start    uint64
+	Duration uint64
+	Stats    []groupedStats
+}
+
+type groupedStats struct {
```

Comment:
Why not use the struct generated by the protobuf of the agent here?
We need to use that struct to send data to the agent anyway using message pack right?

Reply:
It needs to be unexported (not public API) and we only use msgpack. I don't want to share code with the agent, that's too big of a dependency.